### PR TITLE
rust 1.78.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

File: Formula/r/rust.rb [hunk 1]
```diff
@@ -2,17 +2,16 @@ class Rust < Formula
   desc "Safe, concurrent, practical language"
   homepage "https://www.rust-lang.org/"
   license any_of: ["Apache-2.0", "MIT"]
-  revision 1
 
   stable do
     # TODO: On 1.78.0 release, switch to `llvm` 18.
```
Comment (fxcoudert) [inline Formula/r/rust.rb:7]: Should we switch to llvm 18?

Comment (chenrui333) [reply Formula/r/rust.rb:7]: yes! good catch

**Comment by chenrui333** [comment]
- https://github.com/rust-lang/rust/pull/122178
- https://github.com/Homebrew/homebrew-core/pull/169373

**Review by carlocab — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by chenrui333** [comment]
build error list, I have addressed some and we can address the others as followup.
```
brew install --build-from-source flux
brew install --build-from-source git-interactive-rebase-tool
brew fetch --build-from-source --retry gitu
brew fetch --build-from-source --retry gnuradio boost cppzmq gsl dbus libxfixes libxi libxtst at-spi2-core gsettings-desktop-schemas gtk+3 berkeley-db@5 aften jack log4cpp portaudio gobject-introspection py3cairo pygobject3 qt@5 pyqt@5 qwt-qt5 librtlsdr soapysdr soapyrtlsdr fmt spdlog uhd cpu_features orc volk
brew install --build-from-source influxdb
brew install --build-from-source influxdb@1
brew install --build-from-source jupyterlab
brew install --build-from-source kapacitor
brew install --build-from-source librespot
brew install --build-from-source rubyfmt
brew install --build-from-source sapling
brew install --build-from-source scarb
brew install --build-from-source smartdns
brew install --build-from-source solo2-cli
brew install --build-from-source sonic
brew install --build-from-source wagyu
brew install --build-from-source watchman
brew install --build-from-source zet
```

**Comment by carlocab** [comment]
Oops, still running...

**Comment by chenrui333** [comment]
yeah, the 12 is still running 😅 (due to the 12 runner performance, we might want to cancel it, but I will give it a day)

**Pull request merged**